design/crd-configstate.md

```diff
@@ -0,0 +1,288 @@
+spec:
+  myASN: 64512
+  peerASN: 64513
+  peerAddress: 192.168.1.1
+  bfdProfile: my-bfd-profile  # References profile that doesn't exist
+```
+
+**Then** speaker/configReconciler fails to load configuration
+
+   ```bash
+   kubectl apply -f bgppeer.yaml
+   # Output: bgppeer.metallb.io/peer1 created
+
+   kubectl logs -n metallb-system -l component=speaker -c speaker --tail=50 | grep "error"
+   # Output: {"caller":"config_controller.go:140","controller":"ConfigReconciler","error":"peer peer1 referencing non existing bfd profile my-bfd-profile","level":"error"}
+   ```
+
+### Other Validation Errors
+
+Other validation errors can occur for example when the secret exists
+but has incorrect content. The user must fix the secret to resolve these
+errors. The webhook strips the `passwordSecret` field to avoid accessing secret
+content during validation. The reconciler validates the actual secret content.
+
+- Secret Type Mismatch
+  - When: BGPPeer references a secret that exists but has wrong type (not `kubernetes.io/basic-auth`)
+  - Code: [internal/config/config.go:500](https://github.com/metallb/metallb/blob/main/internal/config/config.go#L500)
+
+- Password Field Missing
+  - When: BGPPeer references a secret that exists but doesn't have a `password` field
+  - Code: [internal/config/config.go:505](https://github.com/metallb/metallb/blob/main/internal/config/config.go#L505)
+
+- IPv6 Pool with BFD Echo
+  - When: Pool has IPv6 CIDR, BGPAdvertisement references BGPPeer with BFDProfile that has echo mode enabled
+  - Code: [internal/config/validation.go](https://github.com/metallb/metallb/blob/main/internal/config/validation.go)
+
+**Given** a MetalLB installation in namespace `metallb-system`
+
+**When** user applies a Secret with wrong type and BGPPeer that references it:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bgp-password
+  namespace: metallb-system
+type: Opaque  # Wrong type - should be kubernetes.io/basic-auth
+stringData:
+  password: "mypassword123"
+---
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: peer-with-secret
+  namespace: metallb-system
+spec:
+  myASN: 64512
+  peerASN: 64513
+  peerAddress: 192.168.1.2
+  passwordSecret:
+    name: bgp-password
+```
+
+**Then** speaker/configReconciler fails to load configuration
+
+   ```bash
+   kubectl apply -f secret-and-peer.yaml
+   # Output:
+   # secret/bgp-password created
+   # bgppeer.metallb.io/peer-with-secret created
+
+   kubectl logs -n metallb-system -l component=speaker -c speaker --tail=50 | grep "error"
+   # Output: {"caller":"config_controller.go:140","controller":"ConfigReconciler","error":"parsing peer peer-with-secret secret type mismatch on \"metallb-system\"/\"bgp-password\", type \"kubernetes.io/basic-auth\" is expected \nfailed to parse peer peer-with-secret password secret","level":"error"}
+   ```
+
+
+//TODO configuration errors in speaker/nodecontroller `internal/k8s/controllers/node_controller.go`
```

Comment:
Let's remove / fix the TODOs and I think this can be merged.
There's no need to outline all the possible errors here as long as we cover them in the implementation